### ISSUE TITLE
Bug Issue: QMerge and IMerge failure

### ISSUE THREAD

**Issue by aishwaryachander** [uubram/RTCR#15]
Hi, 
I was using the RTCR pipeline via the DNANexus platform and for fastq.gz processing of paired-end files and came across a "No Such file in Directory" error. As its hosted on a private project, it would be difficult to provide access for the same.

The “Raw clones identified by RTCR before error correction.” step was the last successful step before the pipeline failed. The file r.dat has been generated and we were also able to successfully test the Convert feature and convert the r.dat into human-readable “tsv”. The r.dat file has 104 MB and it is nonempty.

The next pipeline step “QMerge and IMerge” failed. According to the https://github.com/uubram/RTCR/blob/master/docs/README.md, this should generate rqi.dat. The pipeline was creating a pool of workers and computed a merge on a subset of data. One of them failed on its input during the processing. I've attached the lob here for your reference. 

[rtcr.log](https://github.com/uubram/RTCR/files/6207634/rtcr.log)

Any assistance on what the error might possibly mean and if I can do anything differently to access processed files? 
Looking forward to hearing from you!

Regards,
Aishwarya Chander
_Research Assistant
MIODx Inc._

**Comment by uubram** [comment]
According to the error in the log, the child process was killed via SIGKILL. This is probably a result of the system running out of memory and the Out Of Memory (OOM) manager of linux killing one of the child processes of RTCR to free up memory.

After making sure more memory is available, you can restart RTCR (in a new folder with alignments.sam.gz so that it doesn't redo the alignment step). Please let me know if this solves the issue.

**Comment by aishwaryachander** [comment]
Hi, I used a higher memory core and still got the same error, “mv: cannot stat 'data/analysis/rqi.dat': No such file or directory”I was wondering if there might just be something wrong with the input data? (to the effect of an experimental error)Just beginning my journey into bioinformatics so unsure of the technicalities. Could there be any reason other than memory for this error? I’ve run into memory  issues in the past and using a higher memory core has usually fixed it, but it’s failed this time.  Thank you for your prompt response!  Regards,Aishwarya Chander From: uubramSent: 26 March 2021 02:24To: uubram/RTCRCc: aishwaryachander; AuthorSubject: Re: [uubram/RTCR] Bug Issue: QMerge and IMerge failure (#15) According to the error in the log, the child process was killed via SIGKILL. This is probably a result of the system running out of memory and the Out Of Memory (OOM) manager of linux killing one of the child processes of RTCR to free up memory.After making sure more memory is available, you can restart RTCR (in a new folder with alignments.sam.gz so that it doesn't redo the alignment step). Please let me know if this solves the issue.—You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or unsubscribe.

**Comment by uubram** [comment]
I do not think that there is something wrong with your input data, but that you're still running out of memory. The error that 'rqi.dat' was not found, is probably a result of an RTCR child process being killed by the OOM manager. Please check the log to see if you again see that a child process got killed via SIGINT. If a child process was killed, it is likely that more memory is needed for this particular RTCR run.
